allensdk/test/brain_observatory/behavior/data_objects/test_cell_specimens.py

```diff
@@ -91,24 +92,101 @@ def setup_class(cls):
             timestamps=np.array([.1, .2, .3]), number_of_frames=3)
 
     def test_from_json(self):
-        csp = CellSpecimens.from_json(dict_repr=self.dict_repr,
-                                      ophys_timestamps=self.ophys_timestamps)
+        csp = CellSpecimens.from_json(
+            dict_repr=self.dict_repr,
+            ophys_timestamps=self.ophys_timestamps,
+            segmentation_mask_image_spacing=(.78125e-3, .78125e-3))
         assert not csp.table.empty
         assert not csp.events.empty
         assert not csp.dff_traces.empty
         assert not csp.corrected_fluorescence_traces.empty
         assert csp.meta == self.expected_meta
 
+    @pytest.mark.parametrize('data',
+                             ('dff_traces',
+                              'corrected_fluorescence_traces',
+                              'events'))
+    def test_roi_data_same_order_as_cell_specimen_table(self, data):
+        """tests that roi data are in same order as cell specimen table"""
+        csp = CellSpecimens.from_json(
+            dict_repr=self.dict_repr,
+            ophys_timestamps=self.ophys_timestamps,
+            segmentation_mask_image_spacing=(.78125e-3, .78125e-3))
+        private_attr = getattr(csp, f'_{data}')
+        public_attr = getattr(csp, data)
+
+        # Events stores cell_roi_id as column whereas traces is index
+        data_cell_roi_ids = getattr(
+            private_attr.value,
+            'cell_roi_id' if data == 'events' else 'index').values
+
+        current_order = np.where(data_cell_roi_ids ==
+                                 csp._cell_specimen_table['cell_roi_id'])[0]
+
+        # make sure same order
```

Comment:
It's not clear to me what lines 126 through 131 are doing. It looks like you are rearranging the rows in the private_attr._value and public_attr.value by hand. What is the reason for that?

Reply:
Basically I'm testing that regardless of the order in the private accessors, that the public property returns in the same order. 
So it
1) arranges in same order as cell_specimen_table
2) rearranges
3) checks that public property maintains same order as cell_specimen_table

It's sort of a silly test since the public property by design returns in the same order since it does a join with cell_specimen_table

When accessed from the private accessor, or when used outside of `CellSpecimens`, events, traces are not guaranteed to be in the same order

Reply:
Thanks for the explanation. That makes sense.